src/jit/hwintrinsichelperlistxarch.h

```diff
@@ -0,0 +1,32 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+/*****************************************************************************/
+#ifndef SOFTWARE_INTRINSIC
+#error Define SOFTWARE_INTRINSIC before including this file
+#endif
+/*****************************************************************************/
+
+// clang-format off
+
+#ifdef FEATURE_HW_INTRINSICS
+/* Note
+1) Each hardware intrinsic has a unique Intrinsic ID with type of `enum NamedIntrinsic`
+2) All the overloads of an intrinsic in an ISA class share one Intrinsic ID
+3) The helper intrinsic which is implemented in software with use of hardware intrinsics and is part of public API is defined in this table
+4) The proper hardware intrinsics which are implemented in RyuJIT are defined in 'hwintrinsiclistxarch.h' file
+*/
+// ************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************
+//                  Intrinsic ID                                      Function name                                      ISA        ival       SIMD size       NumArg                                                                   instructions                                                                                                Category                                         Flags
+//                                                                                                                                                                            {TYP_BYTE,      TYP_UBYTE,       TYP_SHORT,     TYP_USHORT,    TYP_INT,       TYP_UINT,      TYP_LONG,      TYP_ULONG,     TYP_FLOAT,     TYP_DOUBLE}
+// ************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************
+//  SSE Intrinsics
+SOFTWARE_INTRINSIC(SSE_SetAllVector128,                              "SetAllVector128",                                  SSE,        -1,           16,          1,            {INS_invalid,   INS_invalid,   INS_invalid,   INS_invalid,   INS_invalid,      INS_invalid,   INS_invalid,   INS_invalid,   INS_invalid,   INS_invalid},        HW_Category_Helper,                            HW_Flag_NoFlag)
```

Comment:
You could probably define a helper macro that doesn't require you to fill out all 10 instruction fields (which will always be invalid).

Reply:
If `SOFTWARE_INTRINSIC` macro does not define all `HWIntrinsicInfo` members compiler raises the following warnings and errors:

```
e:\src\ms\dotnet\coreclr-sse2\src\jit\hwintrinsichelperlistxarch.h(25): warning C4003: not enough actual parameters for macro 'SOFTWARE_INTRINSIC' (compiling source file E:\src\ms\dotnet\coreclr-sse2\
src\jit\hwintrinsicxarch.cpp) [E:\src\ms\dotnet\coreclr-sse2\bin\obj\Windows_NT.x64.Checked\src\jit\standalone\clrjit.vcxproj]
e:\src\ms\dotnet\coreclr-sse2\src\jit\hwintrinsichelperlistxarch.h(26): warning C4003: not enough actual parameters for macro 'SOFTWARE_INTRINSIC' (compiling source file E:\src\ms\dotnet\coreclr-sse2\
src\jit\hwintrinsicxarch.cpp) [E:\src\ms\dotnet\coreclr-sse2\bin\obj\Windows_NT.x64.Checked\src\jit\standalone\clrjit.vcxproj]
e:\src\ms\dotnet\coreclr-sse2\src\jit\hwintrinsichelperlistxarch.h(28): warning C4003: not enough actual parameters for macro 'SOFTWARE_INTRINSIC' (compiling source file E:\src\ms\dotnet\coreclr-sse2\
src\jit\hwintrinsicxarch.cpp) [E:\src\ms\dotnet\coreclr-sse2\bin\obj\Windows_NT.x64.Checked\src\jit\standalone\clrjit.vcxproj]
e:\src\ms\dotnet\coreclr-sse2\src\jit\hwintrinsichelperlistxarch.h(29): warning C4003: not enough actual parameters for macro 'SOFTWARE_INTRINSIC' (compiling source file E:\src\ms\dotnet\coreclr-sse2\
src\jit\hwintrinsicxarch.cpp) [E:\src\ms\dotnet\coreclr-sse2\bin\obj\Windows_NT.x64.Checked\src\jit\standalone\clrjit.vcxproj]
C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\IDE\VC\VCTargets\Microsoft.CppCommon.targets(171,5): warning MSB4220: Waiting for the currently executing task "CustomBuild" to ca
ncel. [E:\src\ms\dotnet\coreclr-sse2\bin\obj\Windows_NT.x64.Checked\src\inc\corguids.vcxproj]
e:\src\ms\dotnet\coreclr-sse2\src\jit\hwintrinsichelperlistxarch.h(25): error C2220: warning treated as error - no 'object' file generated (compiling source file E:\src\ms\dotnet\coreclr-sse2\src\jit\
hwintrinsicxarch.cpp) [E:\src\ms\dotnet\coreclr-sse2\bin\obj\Windows_NT.x64.Checked\src\jit\standalone\clrjit.vcxproj]
e:\src\ms\dotnet\coreclr-sse2\src\jit\hwintrinsichelperlistxarch.h(25): error C2059: syntax error: ',' (compiling source file E:\src\ms\dotnet\coreclr-sse2\src\jit\hwintrinsicxarch.cpp) [E:\src\ms\dot
net\coreclr-sse2\bin\obj\Windows_NT.x64.Checked\src\jit\standalone\clrjit.vcxproj]
e:\src\ms\dotnet\coreclr-sse2\src\jit\hwintrinsichelperlistxarch.h(25): error C2143: syntax error: missing ';' before '}' (compiling source file E:\src\ms\dotnet\coreclr-sse2\src\jit\hwintrinsicxarch.
cpp) [E:\src\ms\dotnet\coreclr-sse2\bin\obj\Windows_NT.x64.Checked\src\jit\standalone\clrjit.vcxproj]
e:\src\ms\dotnet\coreclr-sse2\src\jit\hwintrinsichelperlistxarch.h(25): error C2059: syntax error: '}' (compiling source file E:\src\ms\dotnet\coreclr-sse2\src\jit\hwintrinsicxarch.cpp) [E:\src\ms\dot
net\coreclr-sse2\bin\obj\Windows_NT.x64.Checked\src\jit\standalone\clrjit.vcxproj]
e:\src\ms\dotnet\coreclr-sse2\src\jit\hwintrinsichelperlistxarch.h(26): error C2143: syntax error: missing ';' before '{' (compiling source file E:\src\ms\dotnet\coreclr-sse2\src\jit\hwintrinsicxarch.
cpp) [E:\src\ms\dotnet\coreclr-sse2\bin\obj\Windows_NT.x64.Checked\src\jit\standalone\clrjit.vcxproj]
e:\src\ms\dotnet\coreclr-sse2\src\jit\hwintrinsichelperlistxarch.h(26): error C2447: '{': missing function header (old-style formal list?) (compiling source file E:\src\ms\dotnet\coreclr-sse2\src\jit\
hwintrinsicxarch.cpp) [E:\src\ms\dotnet\coreclr-sse2\bin\obj\Windows_NT.x64.Checked\src\jit\standalone\clrjit.vcxproj]
e:\src\ms\dotnet\coreclr-sse2\src\jit\hwintrinsichelperlistxarch.h(26): error C2059: syntax error: ',' (compiling source file E:\src\ms\dotnet\coreclr-sse2\src\jit\hwintrinsicxarch.cpp) [E:\src\ms\dot
net\coreclr-sse2\bin\obj\Windows_NT.x64.Checked\src\jit\standalone\clrjit.vcxproj]
e:\src\ms\dotnet\coreclr-sse2\src\jit\hwintrinsichelperlistxarch.h(28): error C2143: syntax error: missing ';' before '{' (compiling source file E:\src\ms\dotnet\coreclr-sse2\src\jit\hwintrinsicxarch.
cpp) [E:\src\ms\dotnet\coreclr-sse2\bin\obj\Windows_NT.x64.Checked\src\jit\standalone\clrjit.vcxproj]
e:\src\ms\dotnet\coreclr-sse2\src\jit\hwintrinsichelperlistxarch.h(28): error C2447: '{': missing function header (old-style formal list?) (compiling source file E:\src\ms\dotnet\coreclr-sse2\src\jit\
hwintrinsicxarch.cpp) [E:\src\ms\dotnet\coreclr-sse2\bin\obj\Windows_NT.x64.Checked\src\jit\standalone\clrjit.vcxproj]
e:\src\ms\dotnet\coreclr-sse2\src\jit\hwintrinsichelperlistxarch.h(28): error C2059: syntax error: ',' (compiling source file E:\src\ms\dotnet\coreclr-sse2\src\jit\hwintrinsicxarch.cpp) [E:\src\ms\dot
net\coreclr-sse2\bin\obj\Windows_NT.x64.Checked\src\jit\standalone\clrjit.vcxproj]
e:\src\ms\dotnet\coreclr-sse2\src\jit\hwintrinsichelperlistxarch.h(29): error C2143: syntax error: missing ';' before '{' (compiling source file E:\src\ms\dotnet\coreclr-sse2\src\jit\hwintrinsicxarch.
cpp) [E:\src\ms\dotnet\coreclr-sse2\bin\obj\Windows_NT.x64.Checked\src\jit\standalone\clrjit.vcxproj]
e:\src\ms\dotnet\coreclr-sse2\src\jit\hwintrinsichelperlistxarch.h(29): error C2447: '{': missing function header (old-style formal list?) (compiling source file E:\src\ms\dotnet\coreclr-sse2\src\jit\
hwintrinsicxarch.cpp) [E:\src\ms\dotnet\coreclr-sse2\bin\obj\Windows_NT.x64.Checked\src\jit\standalone\clrjit.vcxproj]
e:\src\ms\dotnet\coreclr-sse2\src\jit\hwintrinsichelperlistxarch.h(29): error C2059: syntax error: ',' (compiling source file E:\src\ms\dotnet\coreclr-sse2\src\jit\hwintrinsicxarch.cpp) [E:\src\ms\dot
net\coreclr-sse2\bin\obj\Windows_NT.x64.Checked\src\jit\standalone\clrjit.vcxproj]
e:\src\ms\dotnet\coreclr-sse2\src\jit\hwintrinsicxarch.cpp(29): error C2143: syntax error: missing ';' before '}' [E:\src\ms\dotnet\coreclr-sse2\bin\obj\Windows_NT.x64.Checked\src\jit\standalone\clrji
t.vcxproj]
e:\src\ms\dotnet\coreclr-sse2\src\jit\hwintrinsicxarch.cpp(29): error C2059: syntax error: '}' [E:\src\ms\dotnet\coreclr-sse2\bin\obj\Windows_NT.x64.Checked\src\jit\standalone\clrjit.vcxproj]
```

Reply:
Yes, which is why you define it as:
#define SOFTWARE_INTRINSIC(x, y, z) HARDWARE_INTRINSIC(x, y, INS_invalid, INS_invalid, z)

You then don't need to define the 10 instruction entries (which are always INS_invalid for software intrinsics), but will still have all the required HWIntrinsic Info fields defined